### DISCUSSION TITLE
Authentication in Recovery after using erase-install on M1 Macs

### DISCUSSION THREAD

**Discussion by Flow-loic** [grahampugh/erase-install#85] [Q&A]
Hi,

I've been using this awesome script to automate the reinstall for headless macs in combination with the new MDM feature auto_advance. I just connect to the device, run the script with --erase and --install arguments and then just have to wait till the device is reinstalled, then advances through the screens and loads the profiles and other things and I can then connect again.

This process works flawlessly on Intel Macs, however I have experienced some odd behavior with M1 Macs.
Using a brand new Mac Mini 2020 I was able to run the whole process just like with the Intels. Now if I do the exact same procedure again on the same device it will also install and configure correctly but when I reboot the device, it's booting to recovery and I have to choose an administrator account and then type the password in. After a couple of seconds the authentication is done and the device can boot normally again. Unfortunately I can't do this with headless devices...

Now I'm unsure why this authentication is needed in the first place and why it's not happening with brand new devices. Did anyone get the same behavior? Is there a way around this?

Any help or information on this would be much appreciated!

BR, Loïc

**Comment by grahampugh** [comment]
Hi @Flow-loic are you able to test this out with the latest macOS beta (11.4) I've seen this problem mentioned as resolved in the release notes of macOS 11.4.

**Reply by Flow-loic** [reply]
Hi Graham,
No, I have not tested it with the Beta yet, only the latest official release 11.3.1. I will give it a try and get back!

Thanks and BR!

**Comment by Flow-loic** [comment]
Hi Graham,
So I tested with the Beta of 11.4 (20F5065a) and unfortunately I get the exact same behavior... Runs as supposed the first time but after the second reinstall it starts asking for a administrator password in recovery again. :(

**Reply by grahampugh** [reply]
Yeah, that's confirmed by a couple of users in MacAdmins Slack too. Also, Apple removed the note from the beta release notes for the latest release (the Release Candidate). So, I guess they figured out that they didn't solve it after all.

**Reply by Flow-loic** [reply]
Too bad... but thanks a lot for the information! I guess we'll just have to wait until Apple has resolved this. Let me know if I shall test it again!

**Comment by grahampugh** [comment]
I've added a warning to the Wiki while this bug persists: https://github.com/grahampugh/erase-install/wiki/7.-Erasing-and-reinstalling-macOS#bug-warning

**Reply by Flow-loic** [reply]
Hi Graham,
Since macOS 11.5 came out last week, I gave it another try and... It works! I can now reinstall apple silicon based macs without the device booting to recovery to enter admin credentials. Thanks again for your efforts!
BR
